rewrite-maven/src/main/java/org/openrewrite/maven/UpdateMavenProjectPropertyJavaVersion.java

```diff
@@ -92,13 +92,19 @@ public Xml.Document visitDocument(Xml.Document document, ExecutionContext ctx) {
                     for (Plugin plugin : getResolutionResult().getParent().getPom().getPlugins()) {
                         if (plugin.getGroupId().equals("org.apache.maven.plugins") && plugin.getArtifactId().equals("maven-compiler-plugin") && plugin.getConfiguration() != null) {
                             for (String property : JAVA_VERSION_PROPERTIES) {
-                                if (getResolutionResult().getPom().getProperties().get(property) != null && Float.parseFloat(getResolutionResult().getPom().getProperties().get(property)) < version &&
-                                    (plugin.getConfiguration().get("source") != null && plugin.getConfiguration().get("source").textValue().contains(property)) ||
-                                    (plugin.getConfiguration().get("target") != null && plugin.getConfiguration().get("target").textValue().contains(property)) ||
-                                    (plugin.getConfiguration().get("release") != null && plugin.getConfiguration().get("release").textValue().contains(property))) {
-                                    d = (Xml.Document) new AddPropertyVisitor(property, String.valueOf(version), null)
-                                            .visitNonNull(d, ctx);
-                                    maybeUpdateModel();
+                                if (getResolutionResult().getPom().getProperties().get(property) != null) {
+                                    try {
+                                        int parsed = Integer.parseInt(getResolutionResult().getPom().getProperties().get(property));
```

Comment:
The idea behind using `Float.parseFloat()` is probably to also allow for e.g. `1.5`. I am however also wondering why the `${java.version}` didn't get resolved properly and why it attempted to parse that as-is.

Reply:
Should we try recursive resolution as long as we get back another property? Or should we allow setting the property's value rather than replacing the property reference?

Reply:
@Laurens-W Just so we don't forget about this just because the PR is merged: Do you see anything else here that needs to done? It seems like we probably have a utility somewhere to resolve dependencies recursively.

Reply:
@knutwannheden thanks for the reminder! I think we can and we should recursively check properties for both this new case and the main case of the recipe